### PR TITLE
feat(patterns/modals): add modal pattern

### PR DESCRIPTION
**Pull request by tiloyi**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

Add Modal Pattern

GitHub issue number or Jira issue URL: N/A

### CODE REVIEW

**Comment by caqueste** [comment]
Deployment link → https://featmodaldev-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/Components/Modals/

File: src/docs/Components/Modals/code.mdx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+<Preview path="default" />
+
+## Variations
+
+### Without heading
+
+In cases where it is not necessary, you can use a modal without heading, as follows:
+
+<Preview path="withouttitle" />
```
Comment (nherwegh) [inline src/docs/Components/Modals/code.mdx:77]: modals can't have two primary buttons

File: src/docs/Components/Modals/code.mdx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+### Footer - two call to actions
+
+A primary call to action button accompanied with a secondary button to give the user an alternative
```
Comment (nherwegh) [inline src/docs/Components/Modals/code.mdx:81]: maybe finish the sentence with "alternative choice."

File: src/docs/Components/Modals/code.mdx (hunk 1 above)
Comment (tiloyi) [reply src/docs/Components/Modals/code.mdx:85]: My bad. It's fixed.

**Comment by nherwegh** [comment]
@tiloyi no "e" in every Modal Title preview 😉

File: src/docs/Components/Modals/code.mdx (hunk 2 above)
Comment (tiloyi) [reply src/docs/Components/Modals/code.mdx:89]: fixed

**Comment by tiloyi** [comment]
> @tiloyi no "e" in every Modal Title preview 😉

fixed ;)

**Comment by nherwegh** [comment]
@tiloyi sorry about that but call to actions must take 100% of the modal's width on `breakpoint-s` as shown [here](https://share.goabstract.com/78533af2-8f38-4e31-b3d1-b6dbf88effb3?sectionId=273f281d-7e32-4043-87a6-88a0105abd30). I just fixed it after a feedback my bad, sorry!

**Comment by tiloyi** [comment]
> @tiloyi sorry about that but call to actions must take 100% of the modal's width on `breakpoint-s` as shown [here](https://share.goabstract.com/78533af2-8f38-4e31-b3d1-b6dbf88effb3?sectionId=273f281d-7e32-4043-87a6-88a0105abd30). I just fixed it after a feedback my bad, sorry!

Fixed

File: src/docs/Components/Modals/code.mdx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+A primary call to action button accompanied with a secondary button to give the user an alternative choice.
+
+<Preview path="cta-two" />
+
+### Footer - call to action and link
+
+A primary call to action button with a link to redirect the user to a page to give more informations.
+
+<Preview path="cta-link" />
```
Comment (nherwegh) [inline src/docs/Components/Modals/code.mdx:107]: the link should be centered on small breakpoint

**Comment by tiloyi** [comment]
> You can add a part Javascript needs and accessibility

fixed

**Comment by tiloyi** [comment]
> @tiloyi The overlay between the page and the modal is missing.
> The properties are G900 with 70% of opacity with 1 of the background blur effect.

fixed but not for the "blur effect" because I don't have the value of this effect in the abstract file.

File: src/docs/Components/Modals/code.mdx (hunk 3 above)
Comment (tiloyi) [reply src/docs/Components/Modals/code.mdx:107]: fixed

**Comment by nherwegh** [comment]
> > @tiloyi The overlay between the page and the modal is missing.
> > The properties are G900 with 70% of opacity with 1 of the background blur effect.
> 
> fixed but not for the "blur effect" because I don't have the value of this effect in the abstract file.

maybe:
backdrop-filter: blur(1px);

**Pull request merged**